proxy/client.go

```diff
@@ -57,10 +61,13 @@ func (c *Client) newRequest(method, path string, body io.Reader) (*http.Request,
 	if err != nil {
 		return nil, err
 	}
-	rel := &url.URL{Path: u.Path, RawQuery: u.RawQuery}
-	url := c.GatewayURL.ResolveReference(rel)
+	// deep copy gateway url and then add the supplied path  and args to the copy so that
+	// we preserve the original gateway URL as much as possible
+	endpoint, _ := url.Parse(c.GatewayURL.String())
```

Comment:
Could you change this to not throw away any error raised?

Reply:
we can return the error from here, but it should be impossible because `c.GatewayURL` is already a valid URL object, so `c.GatewayURL.String()` should always be a valid url 

better safe than sorry